### ISSUE TITLE
Lost lines from output

### ISSUE THREAD

**Issue by fpawel** [jstemmer/go-junit-report#123]
This 
[test_out.txt](https://github.com/jstemmer/go-junit-report/files/7039373/test_out.txt)
 contains the output from which the lines were lost.
This is the report generated by go-junit-report 
[report.txt](https://github.com/jstemmer/go-junit-report/files/7039389/report.txt)
In output:
```
=== RUN   TestZpoolTestSuite/TestGetFreeing_zero
    suite.go:63: test panicked: failed
        goroutine 97 [running]:
        runtime/debug.Stack(0xc00007bb48, 0xd2f220, 0x106f140)
                /home/fpawel/sdk/go1.16.5/src/runtime/debug/stack.go:24 +0x9f
        github.com/stretchr/testify/suite.failOnPanic(0xc0006c2600)
                /home/fpawel/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:63 +0x5b
        panic(0xd2f220, 0x106f140)
                /home/fpawel/sdk/go1.16.5/src/runtime/panic.go:965 +0x1b9
        szs-op/zfs.(*ZpoolTestSuite).SetupTest(0xc00021a0c0)
                /home/fpawel/git/szs-op/zfs/zpool_test.go:25 +0x39
        github.com/stretchr/testify/suite.Run.func1(0xc0006c2600)
                /home/fpawel/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:148 +0x66f
        testing.tRunner(0xc0006c2600, 0xc000165710)
                /home/fpawel/sdk/go1.16.5/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
                /home/fpawel/sdk/go1.16.5/src/testing/testing.go:1238 +0x2b3
```
It was parsed like this:
```
<testcase classname="zfs" name="TestZpoolTestSuite/TestGetFreeing_zero" time="0.000">
			<failure message="Failed" type="">    suite.go:63: test panicked: failed&#xA;        goroutine 97 [running]:&#xA;        runtime/debug.Stack(0xc00007bb48, 0xd2f220, 0x106f140)&#xA;        github.com/stretchr/testify/suite.failOnPanic(0xc0006c2600)&#xA;        panic(0xd2f220, 0x106f140)&#xA;        szs-op/zfs.(*ZpoolTestSuite).SetupTest(0xc00021a0c0)&#xA;        github.com/stretchr/testify/suite.Run.func1(0xc0006c2600)&#xA;        testing.tRunner(0xc0006c2600, 0xc000165710)&#xA;        created by testing.(*T).Run</failure>
		</testcase>
```
The following lines from the output were not included in the xml report:
```
                /home/fpawel/sdk/go1.16.5/src/runtime/debug/stack.go:24 +0x9f
                /home/fpawel/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:63 +0x5b
                /home/fpawel/sdk/go1.16.5/src/runtime/panic.go:965 +0x1b9
                /home/fpawel/git/szs-op/zfs/zpool_test.go:25 +0x39
                /home/fpawel/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:148 +0x66f
                /home/fpawel/sdk/go1.16.5/src/testing/testing.go:1193 +0xef
                /home/fpawel/sdk/go1.16.5/src/testing/testing.go:1238 +0x2b3
```

**Comment by lotos2512** [comment]
i have to the problem

**Comment by crockeo** [comment]
As far as I can tell the issue is that the missing lines match [regexOutput](https://github.com/jstemmer/go-junit-report/blob/d5de2960fd1723d55ee015d9be023e6e6de610db/parser/parser.go#L69) (as expected) and only adding this content to `test.Output`, but it's [being overwritten by buffers[cur]](https://github.com/jstemmer/go-junit-report/blob/d5de2960fd1723d55ee015d9be023e6e6de610db/parser/parser.go#L204) when the parser encounters a status line. I'm still trying to wrap my head around the parser, but I think you can fix this by just adding:

```go
buffers[cur] = append(buffers[cur], matches[2])
```

Testing locally that passes tests and, when run against the provided test output:

```xml
<failure message="Failed" type="">
    suite.go:63: test panicked: failed
        goroutine 97 [running]:
        runtime/debug.Stack(0xc00007bb48, 0xd2f220, 0x106f140)
/home/fpawel/sdk/go1.16.5/src/runtime/debug/stack.go:24 +0x9f
        github.com/stretchr/testify/suite.failOnPanic(0xc0006c2600)
/home/fpawel/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:63 +0x5b
        panic(0xd2f220, 0x106f140)
/home/fpawel/sdk/go1.16.5/src/runtime/panic.go:965 +0x1b9
        szs-op/zfs.(*ZpoolTestSuite).SetupTest(0xc00021a0c0)
/home/fpawel/git/szs-op/zfs/zpool_test.go:25 +0x39
        github.com/stretchr/testify/suite.Run.func1(0xc0006c2600)
/home/fpawel/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:148 +0x66f
        testing.tRunner(0xc0006c2600, 0xc000165710)
/home/fpawel/sdk/go1.16.5/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
/home/fpawel/sdk/go1.16.5/src/testing/testing.go:1238 +0x2b3
</failure>
```

Note that indentation of the `/home/*` lines is messed up because the `regexOutput` branch strips whitespace, but it's better than before.

**Comment by jstemmer** [comment]
Thanks for investigating @crockeo and coming up with a potential fix. As you may have noticed, the parser code has evolved into something that's quite difficult to follow and even small changes may have unintended consequences (which may not always be covered by tests).

I don't intend to make any significant changes or accept PR's for the v1 code. I have been working on v2 for a while now, which fixes many outstanding issues (like this one). While not finished yet, I do think it's already an improvement over the current version and I'm planning to release a beta version pretty soon.

**Comment by crockeo** [comment]
I did notice, it took me the morning to wrap my head around what was going on. I'm glad to hear that you're working on improving the parser! I scanned the gotest parser on the v2 version and it is much more readable. Thanks for your work here!

We may just fork this repo for the time being and backport a fix for this specific issue. Looking forward to v2. 😃

**Comment by jstemmer** [comment]
This has been fixed in v2.0.0, which is now released.
